src/views/teacherregistration/teacherregistration.jsx

```diff
@@ -0,0 +1,112 @@
+            formData: {},
+            registrationError: null,
+            step: 0,
+            waiting: false
+        };
+    },
+    advanceStep: function (formData) {
+        formData = formData || {};
+        this.setState({
+            step: this.state.step + 1,
+            formData: defaults({}, formData, this.state.formData)
+        });
+    },
+    register: function (formData) {
+        this.setState({waiting: true});
+        api({
+            host: '',
+            uri: '/classes/register_educator/',
+            method: 'post',
+            useCsrf: true,
+            formData: {
+                username: this.state.formData.user.username,
+                email: formData.user.email,
+                password: this.state.formData.user.password,
+                birth_month: this.state.formData.user.birth.month,
+                birth_year: this.state.formData.user.birth.year,
+                gender: (
+                    this.state.formData.user.gender === 'other' ?
+                    this.state.formData.user.genderOther :
+                    this.state.formData.user.gender
+                ),
+                country: this.state.formData.user.country,
+                is_robot: this.state.formData.user.isRobot,
+                first_name: this.state.formData.user.name.first,
+                last_name: this.state.formData.user.name.last,
+                phone_number: this.state.formData.phone.national_number,
+                organization_name: this.state.formData.organization.name,
+                organization_title: this.state.formData.organization.title,
+                organization_type: this.state.formData.organization.type,
+                organization_other: this.state.formData.organization.other,
+                organization_url: this.state.formData.organization.url,
+                address_country: this.state.formData.address.country,
+                address_line1: this.state.formData.address.line1,
+                address_line2: this.state.formData.address.line2,
+                address_city: this.state.formData.address.city,
+                address_state: this.state.formData.address.state,
+                address_zip: this.state.formData.address.zip,
+                how_use_scratch: this.state.formData.useScratch
+            }
+        }, function (err, res) {
+            this.setState({waiting: false});
+            if (err) return this.setState({registrationError: err});
+            if (res[0].success) return this.advanceStep(formData);
+            this.setState({registrationError: res[0].msg});
+        }.bind(this));
+
+    },
+    render: function () {
+        return (
+            <Deck className="teacher-registration">
+                {this.state.registrationError ?
+                    <Steps.RegistrationError {... this.state} />
+                :
+                    <Progression {... this.state}>
+                        <Steps.UsernameStep     onNextStep={this.advanceStep}
```

Comment:
Another style inconsistency here. Curious to hear if you have any thoughts on how to unify this.